### PR TITLE
fix: now WAF bypass token header is forwarded

### PR DESCRIPTION
**Pull request by nickytonline**
Fixes an issue when the web application firewall (WAF) is active.

Now if the source image being loaded is a local image, the `X-Nf-Waf-Bypass-Token` header is passed so that the image on the origin server can be loaded for transformation.

There is nothing to test in this PR aside from ensuring the tests are all green.

Closes https://github.com/netlify/pod-ecosystem-frameworks/issues/592

### CODE REVIEW

**Comment by nickytonline** [comment]
It looks like the ci step times out for tests. Ran it locally as well and it times out there too. Looks like my change is causing the timeout. 🤔

**Comment by nickytonline** [comment]
I'm going to take it for a test drive by linking it to the next-runtime and then link the next-runtime in a test project. 😅

File: src/index.ts [hunk 1]
```diff
@@ -51,7 +55,7 @@ export function createIPXHandler ({
   responseHeaders,
   localPrefix,
   ...opts
-}: IPXHandlerOptions = {}) {
+}: IPXHandlerOptions = {}, loadSourceImage = defaultLoadSourceImage) {
```
Comment (ascorbic) [inline src/index.ts:58]: Is this just to allow testing?

**Review by ascorbic — APPROVED** [review]
Looks good! Presumably it's not a risk to leak the existence of the token

**Comment by nickytonline** [comment]
> Looks good! Presumably it's not a risk to leak the existence of the token

I don't think so. Thoughts on this @merlyn-at-netlify?

File: src/index.ts (hunk 1 above)
Comment (nickytonline) [reply src/index.ts:58]: Yeah, that was the main reason, but we were already importing `loadSourceImage`, so why not compose instead.

File: test/index.test.ts [hunk 2]
```diff
@@ -96,3 +100,86 @@ test('source image cache pruning', async (t) => {
...
+      multiValueQueryStringParameters: {},
+      multiValueHeaders: {},
+      isBase64Encoded: false,
+      body: null
+    },
+    getHandlerContext()
+  )
+})
+
+test('should not add WAF headers to local images being transformed', async (t) => {
```
Comment (pieh) [inline test/index.test.ts:132]: not sure if I read this right, but I think this should be:
```suggestion
test('should not add WAF headers to remote images being transformed', async (t) => {
```

**Review by pieh — APPROVED** [review]

**Review by merlyn-at-netlify — APPROVED** [review]
Looks harmless to me. No added security concern from forwarding this header.

**Pull request merged**